Sources/Rendering/WebGPU/TextureManager/index.js

```diff
@@ -81,10 +106,11 @@ function vtkWebGPUTextureManager(publicAPI, model) {
       height: req.height,
       depth: req.depth,
       format: req.format,
+      usage: req.usage,
     });
 
     // fill the texture if we have data
-    if (req.nativeArray || req.image) {
+    if (req.nativeArray || req.image || req.canvas) {
```

Comment:
Do we also need to handle `req.jsImageData` here? (Same with the `writeImageData` changes.)

Reply:
req.nativeArray is set for the req.jsImageData case so that is what catches it in this case. I'll take a look at writeImageData because that can be called directly so I probably need to add a nativeArray setting there.